### ISSUE TITLE
[bug]: All add/edit operations very slow in v0.13.1-dev

### ISSUE THREAD

**Issue by dmikushin** [makeplane/plane#2341]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Migrated from v0.10-dev to v0.13.1-dev

All operations on the issues, such as comment, reaction, new issue, etc. have become very slow. The window remains busy at "Adding..." button for a ~minute after a button click. Finally, a red popup message comes with "Update failed". However the update goes through, as it is shown after the window refresh.

There is an issue either with the database or the frontend.

### Steps to reproduce

Do any operation on the issues, such as comment, reaction, new issue, etc in version v0.13.1-dev.

### Browser

Google Chrome

### Version

Self-hosted

**Comment by dmikushin** [comment]
The following errors are dropped to the browser log when adding a reaction:

```
Failed to load resource: the server responded with a status of 504 ()
_app-63006a975fafbe9b.js:15 Uncaught (in promise) <html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.25.0</center>
</body>
</html>
```

**Comment by dmikushin** [comment]
The following error is dropped to the browser log when the comment finally gets added after a long delay:

```
Failed to load resource: the server responded with a status of 504 ()
```

**Comment by dmikushin** [comment]
The corresponding messages in the docker "proxy" service are the following:

```
proxy            | 2023/10/02 22:50:11 [error] 8#8: *399 upstream timed out (110: Operation timed out) while reading response header from upstream, client: 192.168.192.1, server: , request: "POST /api/workspaces/forge/projects/9deff762-109e-4e23-bbad-464cc9f6493e/issues/fbda7ff5-ebb9-43c4-931b-8a8dbf04bc19/comments/ HTTP/1.1", upstream: "http://192.168.192.5:8000/api/workspaces/forge/projects/9deff762-109e-4e23-bbad-464cc9f6493e/issues/fbda7ff5-ebb9-43c4-931b-8a8dbf04bc19/comments/", host: "plan.mikush.in", referrer: "SITE/forge/projects/9deff762-109e-4e23-bbad-464cc9f6493e/issues?peekIssue=fbda7ff5-ebb9-43c4-931b-8a8dbf04bc19"
proxy            | 192.168.192.1 - - [02/Oct/2023:22:50:11 +0000] "POST /api/workspaces/forge/projects/9deff762-109e-4e23-bbad-464cc9f6493e/issues/fbda7ff5-ebb9-43c4-931b-8a8dbf04bc19/comments/ HTTP/1.1" 504 569 "SITE/forge/projects/9deff762-109e-4e23-bbad-464cc9f6493e/issues?peekIssue=fbda7ff5-ebb9-43c4-931b-8a8dbf04bc19" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36"
```

**Comment by dmikushin** [comment]
Feels as a proxy/websockets problem, but no websockets errors in the browser logs.

**Comment by mr-karan** [comment]
The new update feels pretty broken. Comments don't get saved, any edits to description takes a long time as OP mentions and even then there's a source of truth mismatch. On the side peek view, I am seeing the edited description however when I open the full view, I see the unedited one.

**Comment by Thunderbottom** [comment]
Found a fix for this.

The commit here: https://github.com/makeplane/plane/commit/926d2ae0a07df5a40c7313f35b2047990f1408d5, adds SSL parameters to Redis for production systems, which seems to not be documented anywhere. In the same commit, there's a new flag that sets the `DJANGO_SETTINGS_MODULE` environment variable to `DJANGO_SETTINGS_MODULE="plane.settings.selfhosted"`, which is supposedly for selfhosted deployments, and disables the SSL on Redis.

The issue here is that since the variable isn't defined the environment file (at least wasn't for us), it defaulted to  `DJANGO_SETTINGS_MODULE="plane.settings.production"`, which uses SSL for redis, and caused celery to fail connections to redis for both `apiserver` and `bgworker`. If you enable `DEBUG=1` in `apiserver` container, you'll see the following in the logs:

```python3
ValueError: SSL connection parameters have been provided but the specified URL scheme is redis://. A Redis SSL connection URL should use the scheme rediss://.
```

We've specifically added `DJANGO_SETTINGS_MODULE="plane.settings.selfhosted"` to our `apiserver/.env` file, which causes Django to use the [`apiserver/selfhosted.py`](https://github.com/makeplane/plane/blob/develop/apiserver/plane/settings/selfhosted.py) module instead of [`apiserver/production`](https://github.com/makeplane/plane/blob/develop/apiserver/plane/settings/production.py). This seems to have fixed all the listed issues for us.

While I understand these changes are added to the docker-compose file, these also need to be specified as a breaking change in the changelog, so people that upgrade are aware of such changes and do not end up breaking plane entirely.

**Comment by dmikushin** [comment]
@Thunderbottom , thank you very much for sharing this fix! I confirm that I came across the same issue with redis, and replaced `redis://` with rediss://` in all URL. This is NOT the right thing to do. As you said, the right thing is to add the following line to `apiserver/.env`:

```
DJANGO_SETTINGS_MODULE="plane.settings.selfhosted" to our apiserver/.env
```

Fixes  `v0.13.1-dev` for me, closing.
